application-services/custom/camera-management/appcamera/commands.go

```diff
@@ -0,0 +1,143 @@
+	gotoPresetCommand   = "GotoPreset"
+	streamUriCommand    = "StreamUri"
+
+	moveScaleX = 10
+	moveScaleY = 5
+	zoomScale  = 1
+)
+
+func (app *CameraManagementApp) getProfiles(deviceName string) (ProfilesResponse, error) {
+	profiles, err := app.issueGetCommand(context.Background(), deviceName, "Profiles", struct{}{})
+	if err != nil {
+		return ProfilesResponse{}, errors.Wrapf(err, "failed to issue get Profiles command")
+	}
+
+	val := profiles.Event.Readings[0].ObjectValue
+	js, err := json.Marshal(val)
+	if err != nil {
+		return ProfilesResponse{}, errors.Wrapf(err, "failed to marshal profiles json object")
+	}
+	pr := ProfilesResponse{}
+	err = json.Unmarshal(js, &pr)
+	return pr, err
+}
+
+func (app *CameraManagementApp) queryStreamUri(deviceName, profileToken string) (string, error) {
+	req := StreamUriRequest{ProfileToken: profileToken}
+	cmdResponse, err := app.issueGetCommand(context.Background(), deviceName, streamUriCommand, req)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to issue get StreamUri command")
+	}
+	streamUri, errUri := parseStreamUri(cmdResponse)
+	if errUri != nil {
+		return "", errors.Wrapf(errUri, "failed to get stream Uri from the device %s", deviceName)
+	}
+
+	return streamUri, nil
+}
+
+func (app *CameraManagementApp) doPTZ(deviceName, profileToken string, x, y, zoom float64) (dtosCommon.BaseResponse, error) {
+	trans := ptz.Vector{
+		PanTilt: &onvif.Vector2D{
+			X: x * moveScaleX,
+			Y: y * moveScaleY,
+		},
+	}
+	if zoom != 0 {
+		trans.Zoom = &onvif.Vector1D{
+			X: zoom * zoomScale,
+		}
+	}
+
+	cmd := &ptz.RelativeMove{
+		ProfileToken: onvif.ReferenceToken(profileToken),
+		Translation:  trans,
+	}
+	return app.sendPutCommand(deviceName, relativeMoveCommand, cmd)
+}
+
+func (app *CameraManagementApp) getPresets(deviceName string, profileToken string) (GetPresetsResponse, error) {
+
+	cmd := &ptz.GetPresets{
+		ProfileToken: onvif.ReferenceToken(profileToken),
+	}
+
+	presets, err := app.issueGetCommand(context.Background(), deviceName, "GetPresets", cmd)
+	if err != nil {
+		return GetPresetsResponse{}, errors.Wrapf(err, "failed to issue get presets command")
+	}
+
+	val := presets.Event.Readings[0].ObjectValue
+	js, err := json.Marshal(val)
+	if err != nil {
+		return GetPresetsResponse{}, errors.Wrapf(err, "failed to marshal presets json object")
+	}
+	pr := GetPresetsResponse{}
+	err = json.Unmarshal(js, &pr)
+	return pr, err
+}
+
+func (app *CameraManagementApp) gotoPreset(deviceName string, profile string, preset string) (dtosCommon.BaseResponse, error) {
+
+	cmd := &ptz.GotoPreset{
+		ProfileToken: (*onvif.ReferenceToken)(&profile),
+		PresetToken:  (*onvif.ReferenceToken)(&preset),
+	}
+
+	return app.sendPutCommand(deviceName, gotoPresetCommand, cmd)
+}
+
+func (app *CameraManagementApp) sendPutCommand(deviceName string, commandName string, command interface{}) (dtosCommon.BaseResponse, error) {
+	app.lc.Infof("Sending PUT command: %s, %+v", commandName, command)
```

Comment:
While normally I would agree this is a debug type log, I feel like the reason we had it as info was because it helped to de-mystify the internal EdgeX calls that it was making for the users to understand it better.

@trleasher-intel any comments?

Reply:
OK, That was why it was question and not a request to change it... ;-)